src/huggingface_hub/hf_file_system.py

```diff
@@ -379,7 +387,7 @@ def _ls_tree(
     def glob(self, path, **kwargs):
         # Set expand_info=False by default to get a x10 speed boost
         kwargs = {"expand_info": kwargs.get("detail", False), **kwargs}
-        path = self.resolve_path(path).unresolve()
+        path = self.resolve_path(path, revision=kwargs.get("revision")).unresolve()
```

Comment:
this ensures all the returned paths will start with `path` (glob filters out paths that don't start with the input path)